dialects/linux/dproc.c

```diff
@@ -1518,11 +1552,44 @@ process_proc_map(p, s, ss)
 		efs = sv = 1;
 	    else
 		efs = 0;
-	    if (!efs) {
-		if (HasNFS)
-		    sv = statsafely(fp[6], &sb);
-		else
-		    sv = stat(fp[6], &sb);
+
+	    /* For prcesses in different mount namespace from lsof process,
+	     * stat corresponding files under /proc/[pid]/map_files would follow
+	     * symlinks regardless of namespaces.
+	     */
+	    if (diff_mntns) {
+		char path[MAP_PATH_LENGTH];
+		char addr[ADDR_LENGTH];
+		uint64_t start, end;
+		int ret;
+
+		if (sscanf(fp[0], "%lx-%lx", &start, &end) != 2)
+		    goto stat_directly;
+
+		ret = snprintf(addr, sizeof(addr), "%lx-%lx", start, end);
```

Comment:
Why is it need to sscanf() and then snprintf()?

Reply:
Format of files under /proc/[pid]/maps and /proc/[pid]/map_files/* are a little different, mainly on padding zeros.

Reply:
Thanks for clarification.